### ISSUE TITLE
Please implement PIE801 (prefer-simple-return)

### ISSUE THREAD

**Issue by nth10sd** [astral-sh/ruff#6681]
From [flake8-pie](https://github.com/sbdchd/flake8-pie):

```
Return boolean expressions directly instead of returning True and False.

# error
def main():
    if foo > 5:
        return True
    return False

# error
def main():
    if foo > 5:
        return True
    else:
        return False

# ok
def main():
    return foo > 5
```

```
$ ruff --version
ruff 0.0.285
```

**Comment by charliermarsh** [comment]
Is this the same as https://beta.ruff.rs/docs/rules/needless-bool/?

**Comment by nth10sd** [comment]
It's similar.

However, the following testcase is detected as `PIE801` via `flake8-pie` but not by `ruff`:

```
def f():
    x = "x"
    y = "y"
    if x == "x" and y == "z":
        return True
    return False
```

**Comment by dhruvmanila** [comment]
Currently, the [`needless-bool`](https://beta.ruff.rs/docs/rules/needless-bool/) works for an `if` statement _only_. Here, `return False` is _outside_ the `if` statement and so it isn't detected here. If it would be inside an `else` body, then we would detect it (https://play.ruff.rs/43e800ce-14d3-41f5-9e9e-1d86b4ae5d19). I'm not sure if there's a way to get the next statement without using the entire function body.

**Comment by charliermarsh** [comment]
We do something similar in `convert_for_loop_to_any_all`.

**Comment by charliermarsh** [comment]
These are now picked up by [`needless-bool`](https://docs.astral.sh/ruff/rules/needless-bool/) in preview.
